### PR TITLE
Add CEL rules to AdmissionCheck

### PR DESCRIPTION
**Pull request by IrvingMg**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

It replaces all of the validations executed by webhooks for the `AdmissionCheck` type with CRD validation rules.

#### Which issue(s) this PR fixes:

Relates to #463 

#### Does this PR introduce a user-facing change?

```release-note
Added CRD validation rules to AdmissionCheck.

ACTION REQUIRED: Requires Kubernetes 1.25 or newer
```

### CODE REVIEW

**Review by trasc — COMMENTED** [review]
/lgtm

File: pkg/webhooks/webhooks.go [hunk 1]
```diff
@@ -33,9 +33,5 @@ func Setup(mgr ctrl.Manager) (string, error) {
 		return "ClusterQueue", err
 	}
 
-	if err := setupWebhookForAdmissionCheck(mgr); err != nil {
```
Comment (alculquicondor) [inline pkg/webhooks/webhooks.go:36]: 🥳

File: test/integration/webhook/admissioncheck_test.go [hunk 2]
```diff
@@ -7,7 +7,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/ptr"
```
Comment (alculquicondor) [inline test/integration/webhook/admissioncheck_test.go:9]: Remove the empty line. We generally just use 3 blocks of imports: standard, external, kueue.

**Comment by alculquicondor** [comment]
One nit
/hold
/approve

**Pull request merged**